src/resolvers/resolver.ts

```diff
@@ -1,27 +1,52 @@
-import { AppDataSource } from '../data-source';
+import { CreateUserInput, UserInput, UserOutput } from '../models/user-models';
 import { User } from '../entity/User';
+import { AppDataSource } from '../data-source';
+
+const userRepo = AppDataSource.getRepository(User);
+
+async function validateInput(userData: UserInput) {
+  const passwordRegex = new RegExp(/^(?=.*\d)(?=.*[a-z])(?=.*[A-Z])[0-9a-zA-Z]{8,16}$/);
+  if (passwordRegex.test(userData.password) == false) {
```

Comment:
```suggestion
  if (!passwordRegex.test(userData.password)) {
```

É legal evitar `==` e usar sempre `===`. Ou nesse caso, como simplificação, qualquer coisa diferente de false ele deve jogar o erro.